test/conservation_checker_tests.jl

```diff
@@ -26,173 +25,177 @@ struct TestAtmos{I} <: Interfacer.AtmosModelSimulation
-    @test total_water[end] ==
-          (cc.water.sums.TestAtmos + cc.water.sums.TestOcean + cc.water.sums.TestLand + cc.water.sums.SurfaceStub)[end]
-
-    @test abs((total_energy[end] .- tot_energy_an) / total_energy[end]) < 1e-4
-    @test abs((total_water[end] .- tot_water_an) / total_water[end]) < 1e-4
-
-    # check that all the totals in each checker have the same length
-    for sim in values(model_sims)
-        for checker in values(cc)
-            ccs = checker.sums
-            @test length(ccs.total) == length(getfield(ccs, Symbol(name(sim))))
-        end
-    end
 
-end
 
-@testset "test plot_global_conservation with dummy models" begin
-
-    space = TestHelper.create_space(FT)
-
-    # set up model simulations
-    atmos = TestAtmos((; space = space))
-    land = TestOcean((; space = space))
-    ocean = TestLand((; space = space))
-    ice = SurfaceStub((; area_fraction = Fields.ones(space) .* 0.5))
-    model_sims = (; atmos_sim = atmos, land_sim = land, ocean_sim = ocean, ice_sim = ice)
-
-    # conservation checkers
-    cc = (; energy = EnergyConservationCheck(model_sims), water = WaterConservationCheck(model_sims))
-
-    # coupler fields
-    cf = (;
-        F_radiative_TOA = Fields.ones(space),
-        P_net = Fields.zeros(space),
-        P_liq = Fields.zeros(space),
-        P_snow = Fields.zeros(space),
-        F_turb_moisture = Fields.zeros(space),
-    )
-    @. cf.F_radiative_TOA = 200
-    @. cf.P_liq = -100
-
-    # init
-    cs = Interfacer.CoupledSimulation{FT}(
-        nothing, # comms_ctx
-        nothing, # dates
-        space, # boundary_space
-        cf, # fields
-        nothing, # parsed_args
-        cc, # conservation_checks
-        (Int(0), Int(1000)), # tspan
-        Int(200), # t
-        Int(200), # Δt_cpl
-        (;), # surface_masks
-        model_sims, # model_sims
-        (;), # mode
-        (), # diagnostics
-        (;), # callbacks
-        (;), # dirs
-    )
-
-    tot_energy, tot_water = check_conservation!(cs)
-
-    output_plots = "test_cons_plots/"
-    mkpath(output_plots)
-    plot_global_conservation(
-        cs.conservation_checks.energy,
-        cs,
-        figname1 = output_plots * "energy.png",
-        figname2 = output_plots * "energy_log.png",
-    )
-    plot_global_conservation(
-        cs.conservation_checks.water,
-        cs,
-        figname1 = output_plots * "water.png",
-        figname2 = output_plots * "water_log.png",
-    )
-
-    # test that files exist
-    @test isfile("test_cons_plots/energy.png")
-    @test isfile("test_cons_plots/energy_log.png")
-    @test isfile("test_cons_plots/water.png")
-    @test isfile("test_cons_plots/water_log.png")
-
-    # remove output
-    rm(output_plots; recursive = true)
 
+for FT in (Float32, Float64)
+    global get_field(s::TestAtmos, ::Val{:F_radiative_TOA}) = ones(s.i.space) .* FT.(200)
```

Comment:
I don't love that these are `global` but it was the only way to make them visible within the `testset`s, and at least this is just in the tests